### PR TITLE
Avoid calling GetDiagnostics on compilation during EnC analysis

### PR DESCRIPTION
**Pull request by tmat**
**Customer scenario**

Applying an EnC change in a project containing large method bodies might take a long time (minutes) during which the compiler calculates all semantic diagnostics and VS UI thread is blocked.

**Bugs this fixes:**

https://github.com/dotnet/roslyn/issues/10683

**Workarounds, if any**

None.

**Risk**

Medium. Although I went over the EnC analyzer code to find any code paths that might not be hardened against erroneous code it might happen I missed something. As a result an exception may occur, a non-fatal Watson reported and the debugging session terminated.

**Performance impact**

Improves performance.

**Is this a regression from a previous update?**

Roslyn regressed when compared to pre-Roslyn EnC.

**Root cause analysis:**

We used safer approach of not analyzing broken code when the analyzer was implemented.

**How was the bug found?**

Quite a few customer complains.

### CODE REVIEW

File: src/EditorFeatures/CSharpTest/EditAndContinue/RudeEditStatementTests.cs [hunk 1]
```diff
@@ -4416,7 +4416,7 @@ void F()
         }
 
         // Add corresponding test to VB
-        [WpfFact(Skip = "TODO")]
+        [Fact(Skip = "TODO")]
```
Comment (tmat) [inline src/EditorFeatures/CSharpTest/EditAndContinue/RudeEditStatementTests.cs]: Needs change https://github.com/dotnet/roslyn/pull/18940 to go in first.

File: src/EditorFeatures/CSharpTest/EditAndContinue/RudeEditTopLevelTests.cs [hunk 2]
```diff
@@ -4313,6 +4323,76 @@ public void Insert_ExternConstruct()
                 Diagnostic(RudeEditKind.InsertExtern, "public extern C()", FeaturesResources.constructor));
         }
 
+        [Fact(Skip = "TODO")]
```
Comment (tmat) [inline src/EditorFeatures/CSharpTest/EditAndContinue/RudeEditTopLevelTests.cs]: See https://github.com/dotnet/roslyn/pull/18940

**Comment by ivanbasov** [comment]
:shipit:

File: src/EditorFeatures/CSharpTest/EditAndContinue/RudeEditStatementTests.cs (hunk 1 above)
Comment (gafter) [inline src/EditorFeatures/CSharpTest/EditAndContinue/RudeEditStatementTests.cs]: Could you please change the skip string to `"https://github.com/dotnet/roslyn/pull/18940"`

File: src/EditorFeatures/CSharpTest/EditAndContinue/RudeEditTopLevelTests.cs (hunk 2 above)
Comment (gafter) [inline src/EditorFeatures/CSharpTest/EditAndContinue/RudeEditTopLevelTests.cs]: Same here

**Comment by tmat** [comment]
test windows_debug_vs-integration_prtest  please

**Comment by tmat** [comment]
@MattGertz For ask mode approval.

**Pull request merged**